File: 0880-decoded-string-at-index/0880-decoded-string-at-index.py
class Solution:
    def decodeAtIndex(self, s: str, k: int) -> str:
        size=0
        for i in range(len(s)):
            if s[i].isdigit():
                size=size*int(s[i])
            else:
                size+=1
        for i in range(len(s)-1,-1,-1):
            k%=size
            if (s[i].isdigit()):
                size//=int(s[i])
                k%=size
            else:
                if k==0 or k==size:
                    return s[i]
                size-=1
        return ""

File: 0880-decoded-string-at-index/0880-decoded-string-at-index.java
class Solution {

public String decodeAtIndex(String s, int k) {
        int n = s.length();
        long len = 0;

        for(char c : s.toCharArray()){
            if(Character.isDigit(c)){
                len *= (c - '0');
            }
            else len++;
        }

        for(int i=n-1;i>=0;i--){
            char c = s.charAt(i);
            if(Character.isDigit(c)){
                len = len / (c - '0'); 
                k %= len;
            }
            else {
                if(k == 0 || k == len) return Character.toString(c);
                len --;
            }
        }
        return "";
    }
}